src/validators/index.js

```diff
@@ -0,0 +1,30 @@
+import i18n from '@dhis2/d2-i18n'
+
+const isInteger = (value) => {
+    if (typeof value !== 'string') {
+        return false
+    }
+
+    return (
+        !isNaN(value) &&
+        parseInt(Number(value)) === Number(value) &&
+        !isNaN(parseInt(value, 10))
+    )
+}
+
+export function createIntegerRangeValidator(min, max) {
+    function integerRangeValidator(value) {
+        if (!isInteger(value)) {
+            return false
+        }
+
+        const valueAsInt = parseInt(value)
+
+        return valueAsInt >= min && valueAsInt <= max
+    }
+    integerRangeValidator.message = i18n.t(
+        'This field should contain a round number between {{min}} and {{max}}',
+        { min, max }
+    )
+    return integerRangeValidator
+}
```

Comment:
Perhaps this is a bit of an odd validator. The reason for it being this way is that it gets passed to the legacy `FormBuilder` component which is expecting validator functions which return a boolean and have a `message` property.

Reply:
Couldn't we just use a standard `<input type="number" step="1" min={min} max={max}>` instead of validating in javascript?  In particular I worry that "a round number" might be cryptic to many users - it's not necessarily a universal way to refer to integers

cc @cooper-joe

Reply:
Sorry.... I'm a bit slow to catch on.... I guess your point is that, if we added these attributes, it would be impossible for the user to enter an invalid value. But IMO, then we'd still need some sort of message to explain what the min and max are. So then we still end up having to think of a text to accompany the field. But overall I guess that would be a slightly cleaner fix.

Reply:
@amcgee I've had a think about this and quite liked the idea. My plan was as follows:
- Use the `min` and `max` attribute on the input as you suggested (no need for `step`, 1 is the default)
- Add a help-text below to explain to the user they are allowed to enter integers between 1 and 28
- Get rid of the javascript validation

But then I tried this and wasn't too happy. The issue is with the V0 MUI `TextField` component and showing some sort of helptext. I looked at [the docs](https://v0.mui.com/#/components/text-field) and the component has a prop called `hintText`, which seems to be the right tool for the job. But it turns out to be a placeholder text. So it only shows when this field is empty, which isn't great for our use case, since this field has a default value so the user won't see the hint until he/she actively clears the field.

As such, I think probably the best way forward is to keep using an error-text and javascript validation. I do agree we need to tweak the validation message. Making it generic will make it hard to make it digestible for users, but we can make specific to this usecase if I tweak the implementation slightly.

I've kept the text as-is for now, until we decide on a better text, but I've updated the implementation as I outlined above already in [31a9dc0](https://github.com/dhis2/settings-app/pull/1067/commits/31a9dc09cf565d3dc0e977cbcf3f08146cbeda02).

But I did also notice that while the validation works correctly, it doesn't seem to be blocking the POST if the field is invalid. And the same is true for other pre-existing fields with validation. So I'll see if I can do a quick fix for that tomorrow as well.

Reply:
Agreed that `round`/`whole` number are difficult terms to include in error messages. I think the error(s) could be written as:
- If the input is not a number: `Must be a number.`
- If the input is out of min-max range: `Must be a number between {min} and {max}.`
- If the input is not a whole number: `Must be a whole number without decimal values.`
    -  If we really want to be helpful here, we could generated the rounded up/down values and suggest them. So if the user entered `12.75`, we could say `Must be a whole number without decimal values. Try 12 or 13 instead.` This may not be needed for this edge case though.